src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +48,31 @@ public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final List<String>
     }
   }
 
+  /** Get historical prices for the symbol
+   * @param symbol a symbol to find historical prices for
+   * @param range an optional range to check
+   * @param date an optional date to check
+   */
+  public List<IexHistoricalPrices> getHistoricalPricesForSymbols(final String symbol, final String range, final String date) throws Exception{
 
+    // Input validation
+    if(StringUtils.isBlank(symbol)){ // Symbol is required
+      throw new EntityNotFoundException(symbol.getClass(), symbol);
+    } else if(StringUtils.isBlank(range) && StringUtils.isBlank(date)){ // No range or date specified
+      return iexClient.getHistoricalPricesForSymbol(symbol);
+    } else if ((StringUtils.isBlank(range) == false) && StringUtils.isBlank(date)) { // Range specified but not date
+      if(range == ""){
```

Comment:
isBlank covers the empty scneario for range, this is unnecessary

Reply:
I was having issues with isBlank for date and range since the IEX API response is different if range is an empty string vs when it's not included.

Reply:
made changes so that logic is old. The new commit does this correctly :)